### PR TITLE
pre release: use clones source

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

/kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix
> /kind release

**What this PR does / Why we need it**:

Rather than using the pushed up source from local, the main branch of the repository should be the source of truth for any code that is run and deployed.

**Which issue(s) this PR fixes**:

Hopefully this unblocks release #3402.

**Special notes for your reviewer**:

Also, this should also be done to post_cloudbuild.yaml at a later date.

### CODE REVIEW

**Review by gongmax — APPROVED** [review]

File: build/release/pre_cloudbuild.yaml [hunk 1]
```diff
@@ -13,16 +13,24 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 steps:
+
+  #
+  # Clean out the working dir, so we can replace it.
+  #
+  - name: ubuntu
+    script: rm *
```
Comment (Kalaiselvi84) [inline build/release/pre_cloudbuild.yaml:21]: Could we include `-r` here?

Comment (markmandel) [reply build/release/pre_cloudbuild.yaml:21]: Don't need to - there's only 2 yaml files in there.

**Comment by markmandel** [comment]
Haven't see that flake before

gke-autopilot-1.25:

```
time="2023-09-27 21:36:01.219" level=info msg="<nil>"
    pod_termination_test.go:100: 
        	Error Trace:	/go/src/agones.dev/agones/test/e2e/allocator/helper_func.go:251
        	            				/go/src/agones.dev/agones/test/e2e/allocator/pod_termination_test.go:100
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:222
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:235
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:662
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:596
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:528
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:514
        	            				/go/src/agones.dev/agones/test/e2e/allocator/pod_termination_test.go:97
        	Error:      	Expected value not to be nil.
        	Test:       	TestAllocatorAfterDeleteReplica
    pod_termination_test.go:101: 
        	Error Trace:	/go/src/agones.dev/agones/test/e2e/allocator/pod_termination_test.go:101
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:222
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:235
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:662
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:596
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:528
        	            				/go/src/agones.dev/agones/test/e2e/allocator/wait.go:514
        	            				/go/src/agones.dev/agones/test/e2e/allocator/pod_termination_test.go:97
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unavailable desc = error reading from server: read tcp 192.168.10.2:60758->34.41.84.132:443: read: connection reset by peer
        	Test:       	TestAllocatorAfterDeleteReplica
        	Messages:   	Failed grpc allocation request
```

File: build/release/pre_cloudbuild.yaml [hunk 2]
```diff
@@ -13,16 +13,24 @@
...
   #
   - name: gcr.io/cloud-builders/git
-    dir: /workspace
+    id: clone
     args:
       - clone
       - --depth=1
       - --branch=main
       - https://github.com/googleforgames/agones.git
+      - .
```
Comment (Kalaiselvi84) [inline build/release/pre_cloudbuild.yaml:33]: ./agones works for me

Comment (markmandel) [reply build/release/pre_cloudbuild.yaml:33]: Not 100% sure what you mean here, can you expand?

Comment (Kalaiselvi84) [reply build/release/pre_cloudbuild.yaml:33]: I am receiving an error related to `.` at [this link](https://gist.github.com/Kalaiselvi84/7b4724b8ab4c8b06d65ab8ce46f7643d) and another error related to `script: rm *` at [this link](https://gist.github.com/Kalaiselvi84/c543a5e1d4643fcb4017dc538ad26f7c)

Comment (markmandel) [reply build/release/pre_cloudbuild.yaml:33]: Are you running the entire pull request, or just the changes in this file?

Comment (markmandel) [reply build/release/pre_cloudbuild.yaml:33]: Looking at your error message, you weren't also running with the changes to `release.mk` - is that correct?

Comment (Kalaiselvi84) [reply build/release/pre_cloudbuild.yaml:33]: Yes, I have attempted to execute post_cloudbuild.yaml file directly, not via `post-build-release` target

Comment (Kalaiselvi84) [reply build/release/pre_cloudbuild.yaml:33]: Both steps were completed successfully, and no issues were encountered with pre-build-release target

Comment (markmandel) [reply build/release/pre_cloudbuild.yaml:33]: Brilliant! Let's get this merged, and kickoff the release again.

**Pull request merged**